nixos/modules/services/hardware/amdgpu.nix

```diff
@@ -0,0 +1,40 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.hardware.amdgpu;
+in {
+  options.hardware.amdgpu = {
+    legacySupport.enable = lib.mkEnableOption ''support for old AMD graphics cards, this uses
+      `amdgpu` instead of `radeonsi` driver Southern Islands (SI) and Sea Islands (CIK) cards
+    '';
+    initrd.enable = lib.mkEnableOption ''loading `amdgpu` kernelModule at stage 1.
```

Comment:
Wording nit: kernel module, no need to use NixOS option names here. Also, we usually say "in stage 1", I think?